web_monitoring/tests/test_diffing_server_exc_handling.py

```diff
@@ -0,0 +1,55 @@
+from tornado.testing import AsyncHTTPTestCase
+import web_monitoring.diffing_server as df
+from web_monitoring.diff_errors import UndecodableContentError
+
+class DiffingServerExceptionHandlingTest(AsyncHTTPTestCase):
+
+    def get_app(self):
+        app = df.make_app()
+        app.listen(str(self.get_http_port()))
+        return
+
+    def testInvalidURLaFormat(self):
+        response = self.fetch('http://localhost:'+str(self.get_http_port())+'/html_token?format=json&include=all&a=example.org&b=https://example.org', method='GET')
+        self.assertEqual(response.code, 400)
```

Comment:
We haven’t done a good job of this everywhere, but we are trying to make sure that this server always responds with JSON. It would be great if these tests didn’t only check the response code, but also checked that they:

1. Have the `application/json` content-type
2. Are parse-able JSON
3. The JSON data is an object with at least the fields `code` (a number) and `error` (a string) present

Same for all the other tests here. You could probably write a method that does those assertions and use it in each of the tests.